shopinvader_address/models/res_partner.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import MissingError, UserError
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    def _shopinvader_billing_addresses_already_used(self):
+        """
+        Check if Address is used on posted account move
+        """
+        self.ensure_one()
+        move_id = (
+            self.env["account.move"]
+            .sudo()
+            .search(
+                [("commercial_partner_id", "=", self.id), ("state", "=", "posted")],
```

Comment:
```suggestion
                [("commercial_partner_id", "=", self.id), ("state", "!=", "draft")],
```

Reply:
I think that if a user has only cancelled moves, he should be able to modify his address. 
What about: `"state", "not in", ("draft","cancel")` ?

Reply:
Maybe. Alternatively we could search on confirmed sale orders. Does it make sense for the customer to change the billing address after confirming the order (even if it is not invoiced yet) ?

Reply:
Searching on confirmed sale orders makes more sense to me. It will also allows to remove the dependency on account.

Reply:
Thank you both, I will change it